### ISSUE TITLE
G5Api Stats - Teams swapped on stats

### ISSUE THREAD

**Issue by lanslide-team** [Lan2Play/PugSharp#105]
After the veto, if Team 1 selects switch (opposite team), Team 1 and Team 2 are swapped in the match/player stats feed through to G5API. 
I imagine this is related to the fact, that you have changed Team 1's side, but that isn't reflected when updating G5API.

**Comment by Apfelwurm** [comment]
@lanslide-team on our current feature branch, we sadly can not reproduce that behavior at the moment. I'll report as soon as we merge that stuff so you can try one more time. Maybe we have to then schedule something to look into it together if its still happening

**Comment by lanslide-team** [comment]
I've recorded a video which will hopefully help.

Video - https://www.youtube.com/watch?v=1xP1367QxPM
Config (match.json) - https://pastebin.com/g6JjrBSZ

**Comment by brendontaylor** [comment]
Unfortunately, this still seems to be an issue. Situation to reproduce:

**SETUP**
Team A - CT
- Player 1: PAPA Smurf

Team B - T
- Player 2: Nuke

![image](https://github.com/Lan2Play/PugSharp/assets/14009432/186bb4f3-74f6-46cc-8ed7-b64ca3f7a313)

1. Player 2 [Nuke] starts veto / Player 1 [PAPA smurf] picks the map.
2. After veto Player 2 [Nuke] selects CT as starting side (swap).
3. After the map changes, Player 2 [Nuke] is now CT as "Team A", rather than "Team B".

![image](https://github.com/Lan2Play/PugSharp/assets/14009432/6c78cc44-5f0b-40da-adc5-f442aabc3309)

**Comment by brendontaylor** [comment]
Unfortunately, this still seems to be an issue. Situation to reproduce:

**SETUP**
Team A - CT
- Player 1: PAPA Smurf

Team B - T
- Player 2: Nuke

![image](https://github.com/Lan2Play/PugSharp/assets/14009432/186bb4f3-74f6-46cc-8ed7-b64ca3f7a313)

1. Player 2 [Nuke] starts veto / Player 1 [PAPA smurf] picks the map.
2. After veto Player 2 [Nuke] selects CT as starting side (swap).
3. After the map changes, Player 2 [Nuke] is now CT as "Team A", rather than "Team B".

![image](https://github.com/Lan2Play/PugSharp/assets/14009432/6c78cc44-5f0b-40da-adc5-f442aabc3309)

**Comment by lanslide-team** [comment]
Unfortunately, this still seems to be an issue. Situation to reproduce:

**SETUP**
Team A - CT
- Player 1: PAPA Smurf

Team B - T
- Player 2: Nuke

![image](https://github.com/Lan2Play/PugSharp/assets/14009432/186bb4f3-74f6-46cc-8ed7-b64ca3f7a313)

1. Player 2 [Nuke] starts veto / Player 1 [PAPA smurf] picks the map.
2. After veto Player 2 [Nuke] selects CT as starting side (swap).
3. After the map changes, Player 2 [Nuke] is now CT as "Team A", rather than "Team B".

![image](https://github.com/Lan2Play/PugSharp/assets/14009432/6c78cc44-5f0b-40da-adc5-f442aabc3309)

**Comment by ailiyaaa2030** [comment]
how you can help me to swap?

**Comment by lanslide-team** [comment]
This wasn't an error with the teams swapping, but rather how I had my config file.

On server creation I set Team A as CT (mp_teamname_1) and Team B as T (mp_teamname_2)
The plugin forces the first player to join to be on the Terrorist team as Team 1.
Due to the fact, I didn't set a veto map, the map wasn't restarting after the config was reloaded.

So during the veto I was on Team B based on the scoreboard, but the plugin assigned me to Team 1.
After the map changed it had adjusted the scoreboard, causing the issue.

**tl;dr: Ensure the veto map is set in the match config so the map is reset after the config is loaded**
